src/test/java/com/rarchives/ripme/tst/ripper/rippers/ChanRipperTest.java

```diff
@@ -27,7 +31,22 @@ public void testChanURLPasses() throws IOException, URISyntaxException {
         passURLs.add(new URI("https://rbt.asia/g/thread/70643087/").toURL()); //must work with TLDs with len of 4
         for (URL url : passURLs) {
             ChanRipper ripper = new ChanRipper(url);
-            ripper.setup();
+            // Use CompletableFuture to run setup() asynchronously
+            CompletableFuture<Void> setupFuture = CompletableFuture.runAsync(() -> {
+                try {
+                    ripper.setup();
```

Comment:
This test unit would take forever, and still fail on my machine. I didn't want to modify any behaviors other than just giving up after X time.